examples/sphere/shallow_water.jl

```diff
@@ -105,34 +119,23 @@ else # default case, steady-state test case
     const h0 = 2.94e4 / g
 end
 
-# Plot variables and auxiliary function
-ENV["GKSwstype"] = "nul"
-using ClimaCorePlots, Plots
-Plots.GRBackend()
-dir = "cg_sphere_shallowwater_$(test_name)"
-dir = "$(dir)_$(test_angle_name)"
-path = joinpath(@__DIR__, "output", dir)
-mkpath(path)
-
-function linkfig(figpath, alt = "")
-    # Buildkite-agent upload figpath
-    # Link figure in logs if we are running on CI
-    if get(ENV, "BUILDKITE", "") == "true"
-        artifact_url = "artifact://$figpath"
-        print("\033]1338;url='$(artifact_url)';alt='$(alt)'\a\n")
-    end
-end
-
 # Set up discretization
 ne = 9 # the rossby_haurwitz test case's initial state has a singularity at the pole. We avoid it by using odd number of elements
 Nq = 4
 
 domain = Domains.SphereDomain(R)
 mesh = Meshes.EquiangularCubedSphere(domain, ne)
-grid_topology = Topologies.Topology2D(mesh)
 quad = Spaces.Quadratures.GLL{Nq}()
-space = Spaces.SpectralElementSpace2D(grid_topology, quad)
-
+if usempi
+    grid_topology = Topologies.DistributedTopology2D(context, mesh)
+    global_grid_topology = Topologies.Topology2D(mesh)
+    space = Spaces.SpectralElementSpace2D(grid_topology, quad)
+    global_space = Spaces.SpectralElementSpace2D(global_grid_topology, quad)
+else
+    comms_ctx = nothing
```

Comment:
This block seems to be specific to `Topology` and `Space` assignments, suggest moving the `comms_ctx` variable to the logger block in lines `10-28`. (Is this assignment necessary? I don't see `comms_ctx` being used elsewhere in this example.)

Reply:
done.